site2/docs/client-libraries-java.md

```diff
@@ -607,6 +607,19 @@ consumer 2 will receive:
 ("key-4", "message-4-2")
 ```
 
+By default, consumer in `Key_Shared` subscription will be assigned a fixed hash range of key automatically. If you want to specify the hash ranges of a consumer, you can using the key shared policy:
```

Comment:
```suggestion
By default, a consumer in the key_shared subscription is assigned a fixed hash range of keys automatically. If you want to specify the hash range of a consumer, you can use the key shared policy:
```

Zero conditional sentences express general truths—situations in which one thing always causes another. When you use a zero conditional sentence, you’re talking about a general truth rather than a specific instance of something. 

Example

If you don’t brush your teeth, you get cavities.
When people smoke cigarettes, their health suffers.

Reply:
We'd better keep `Key_Shared`, since other places also use `Key_Shared` and we have already publish the `Key_Shared` subscription in 2.4.0 release. 
A consumer can specify with multiple key hash ranges, is it ok to use `key hash ranges` instead `hash range of keys`

Reply:
1. For key_shared subscription, there are many inconsistencies (issues of capitalization and backtick) in Pulsar doc

* w/o backtick
![image](https://user-images.githubusercontent.com/50226895/69615827-86131300-1070-11ea-82cc-336a86adaf9b.png)

* w/ backtick
![image](https://user-images.githubusercontent.com/50226895/69615747-61b73680-1070-11ea-9e5f-098ea6bf081e.png)

**Shall we use `Key_Shared` subscription overall?**

2. Do you mean changing it to the following?

`By default, a consumer in the key_shared subscription is assigned a fixed hash range of keys automatically. If you want to specify the hash ranges of a consumer, you can use the key shared policy:`

Reply:
> Shall we use Key_Shared subscription overall?
Yes, it's better to use `Key_Shared`,  in the java source code file also use `Key_Shared`

> Do you mean changing it to the following?

By default, a consumer in the key_shared subscription is assigned a fixed key hash range automatically. If you want to specify the key hash ranges of a consumer, you can use the key shared policy:

Reply:
OK, thanks for your explanations.